### DISCUSSION TITLE
Tracking tokens and spend

### DISCUSSION THREAD

**Discussion by kitsamho** [iryna-kondr/scikit-llm#26] [Ideas]
Not sure if this is really out of scope for a library like this - but token usage using OpenAI keys can seriously add up $$. Is there a way we can for each instantiation of an object keep a tally of tokens used as an attribute? Maybe that's a bit clunky.

**Comment by OKUA1** [comment]
We did not have any plans for such a functionality. But we will investigate the possibilities of adding this.
